tests/taskpoller.go

```diff
@@ -452,6 +452,7 @@ retry:
 				Details:   payloads.EncodeString("details"),
 				Identity:  p.Identity,
 			})
+			// todo: we are losing the err2 here
 			return err
```

Comment:
Another example of having 2 different errors in the same scope causing easy-to-miss bugs.

Reply:
This isn't really a bug, ActivityTaskHandler shouldn't return both true for cancel (that would be a success) and an non-nil error (and I just verified this for all implementations). Arguably the if clause above should be `if err2 == nil && cancel`